File: src/main/org/nlogo/window/ExternalFileInterface.java
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.window;

import org.nlogo.api.AgentKind;
import org.nlogo.api.AgentKindJ;

public strictfp class ExternalFileInterface
    implements org.nlogo.api.SourceOwner {

  private final String fileName;

  public ExternalFileInterface(String fileName) {
    this.fileName = fileName;
  }

  public String getFileName() {
    return fileName;
  }

  public String classDisplayName() {
    return "ExternalFileInterface";
  }

  public AgentKind kind() {
    return AgentKindJ.Observer();
  }

  public String headerSource() {
    return "";
  }

  public String innerSource() {
    return "";
  }

  public void innerSource(String s) {
  }

  public String source() {
    return headerSource() + innerSource();
  }

}

File: src/main/org/nlogo/window/ExternalFileInterface.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.window

import org.nlogo.api.{ AgentKind, SourceOwner }

class ExternalFileInterface(fileName: String) extends SourceOwner {
  def getFileName: String = fileName
  def classDisplayName: String = "ExternalFileInterface"
  def kind: AgentKind = AgentKind.Observer
  def headerSource: String = ""
  def innerSource: String = ""
  def innerSource_=(s: String): Unit = {}
  def source: String = headerSource + innerSource
}
